src/main/java/iudx/rs/proxy/apiserver/query/QueryMapper.java

```diff
@@ -198,15 +200,19 @@ private boolean isValidTimeInterval(
     } else if (timeRel.equalsIgnoreCase("before")) {
 
     }
-    if (!isAsyncQuery && totalDaysAllowed > VALIDATION_MAX_DAYS_INTERVAL_ALLOWED) {
-      isValid = false;
-      DxRuntimeException ex =
-          new DxRuntimeException(
-              BAD_REQUEST.getValue(),
-              INVALID_TEMPORAL_PARAM_URN,
-              "time interval greater than 10 days is not allowed");
-      this.context.fail(400, ex);
+    LOGGER.debug("isTimeLimitEnabled : {}", isTimeLimitEnabled);
+    if (isTimeLimitEnabled) {
```

Comment:
i guess it will be better to use single if statement.

Reply:
solved